homeassistant/components/vesync/fan.py

```diff
@@ -157,30 +157,7 @@ def unique_info(self):
     @property
     def extra_state_attributes(self):
         """Return the state attributes of the fan."""
-        attr = {}
-
-        if hasattr(self.smartfan, "active_time"):
-            attr["active_time"] = self.smartfan.active_time
-
-        if hasattr(self.smartfan, "screen_status"):
-            attr["screen_status"] = self.smartfan.screen_status
-
-        if hasattr(self.smartfan, "child_lock"):
-            attr["child_lock"] = self.smartfan.child_lock
-
-        if hasattr(self.smartfan, "night_light"):
-            attr["night_light"] = self.smartfan.night_light
-
-        if hasattr(self.smartfan, "air_quality"):
-            attr["air_quality"] = self.smartfan.air_quality
-
-        if hasattr(self.smartfan, "mode"):
-            attr["mode"] = self.smartfan.mode
-
-        if hasattr(self.smartfan, "filter_life"):
-            attr["filter_life"] = self.smartfan.filter_life
-
-        return attr
+        return self.smartfan.details
```

Comment:
We need to set the keys of the state attributes explicitly so we have control of them.

We're also moving away from using state attributes for all measurements that are relevant on their own. All of these attributes except probably `mode` should preferably be separate sensor entities instead.

Reply:
> We need to set the keys of the state attributes explicitly so we have control of them.

Ahh, that's reasonable. I'll switch it back to being explicit.

> We're also moving away from using state attributes for all measurements that are relevant on their own. All of these attributes except probably mode should preferably be separate sensor entities instead.

This is great to hear from a user perspective, jamming stuff into `attributes` is kind of annoying (I had to make a sensor locally to track the `air_quality` attribute over time). That's going to make this a much bigger change though 😭. I'll see what I can do.

Thanks for the quick feedback!